HierarchicalDataConverter/Catalyst.HierarchicalDataConverter/LoggingHelper2.cs

```diff
@@ -13,13 +13,14 @@ static LoggingHelper2()
             threadlock = new object();
         }
 
-        public static void Debug(Guid guid, string message)
+        public static void Debug(string message)
         {
             try
             {
                 lock (threadlock)
```

Comment:
Have you considered using ReaderWriterLockSlim instead?

Reply:
I had not - I'll look into it, but this should all go away when I add a logging framework setup today or tomorrow

Reply:
Great - logging framework would be a much better approach.